plugins/subspace_storage/dole.ts

```diff
@@ -42,55 +42,67 @@ export class NeuralDole {
 	constructor({ items }: { items: lib.ItemDatabase }) {
 		// Set some storage variables for the dole divider
 		this.items = items;
-		this.itemsLastTick = new Map(items.getEntries());
+		this.itemsLastTick = new Map();
+		for (let [name, qualities] of items.getEntries()) {
+			for (let [quality, count] of Object.entries(qualities)) {
+				this.itemsLastTick.set(name+quality, count);
+			}
+		}
 	}
 
 	doMagic() {
-		for (let [name, count] of this.items.getEntries()) {
-			let magicData = doleNN.tick(
-				count,
-				this.dole[name],
-				this.itemsLastTick.get(name) || 0,
-				this.getRequestStats(name, 5)
-			);
-			this.stats[name] = this.stats[name] || [];
-			this.stats[name].unshift({ req: 0, given: 0 }); // stats[name][0] is the one we currently collect
-			if (this.stats[name].length>10) {
-				this.stats[name].pop(); // remove if too many samples in stats
-			}
+		for (let [name, qualities] of this.items.getEntries()) {
+			for (let [quality, count] of Object.entries(qualities)) {
+				let magicData = doleNN.tick(
+					count,
+					this.dole[name+quality],
+					this.itemsLastTick.get(name+quality) || 0,
+					this.getRequestStats(name, quality, 5)
+				);
+				this.stats[name+quality] = this.stats[name+quality] || [];
+				this.stats[name+quality].unshift({ req: 0, given: 0 }); // stats[name][0] is the one we are collecting
+				if (this.stats[name+quality].length>10) {
+					this.stats[name+quality].pop(); // remove if too many samples in stats
+				}
 
-			this.dole[name] = magicData[0];
-			// DONE handle magicData[1] for graphing for our users
-			prometheusNNDoleGauge.labels(name).set(magicData[1] || 0);
+				this.dole[name+quality] = magicData[0];
+				// DONE handle magicData[1] for graphing for our users
+				prometheusNNDoleGauge.labels(name+quality).set(magicData[1] || 0);
```

Comment:
This is a lot of `name+quality` could this be moved into a variable?

Reply:
The variable would pretty much be name_quality - for performance it doesn't really matter and neither does it for brevity.